module/platform/darwin/SensorFilter/src/SensorFilter.cpp

```diff
@@ -491,43 +514,37 @@ namespace module::platform::darwin {
                                              config.motionFilter.noise.measurement.flatFootOrientation,
                                              MeasurementType::FLAT_FOOT_ORIENTATION());
                     }
+                    // Else the foot is off the ground, so we make sure that for the next time step, we know that
+                    // this time step, the foot was off the ground
                     else if (!foot_down) {
                         previous_foot_down[side] = false;
                     }
 
+                    // Note that the Hwf is set, even if the foot is not down. This means that moving feet in the air
+                    // will have an Hwf associated with them which is the transform from when that foot last hit
+                    // the ground
                     sensors->feet[side].Hwf = footlanding_Hwf[side].matrix();
                 }
 
                 // Calculate our time offset from the last read
-                double deltaT = std::max(
+                // too hard to read
+                const double deltaT = std::max(
                     (input.timestamp - (previousSensors ? previousSensors->timestamp : input.timestamp)).count()
                         / double(NUClear::clock::period::den),
                     0.0);
 
                 // Time update
                 motionFilter.time(deltaT);
 
-                // Gives us the quaternion representation
+                // Convert the motion filter's state vector to a nicer representation, so we can access its elements
                 const auto o = MotionModel<double>::StateVec(motionFilter.get());
 
-                // Map from world to torso coordinates (Rtw)
+                // We make Hwt first, because `o` is in world space
                 Eigen::Affine3d Hwt;
                 Hwt.linear()      = o.Rwt.toRotationMatrix();
-                Hwt.translation() = o.rTWw;
+                Hwt.translation() = -o.rTWw;
                 sensors->Htw      = Hwt.inverse().matrix();
 
-                // Integrate gyro to get angular positions
-                sensors->angular_position = o.omegaTTt / 90.0;
```

Comment:
This is baffling, but the result was never used anyway (besides below, where it logged the rotational "position" as `p_x`, `p_y`, and `p_z`???). Also, that field in the message file had a comment saying that it was deprecated, so I removed it.

Reply:
This is someone getting mad at the odometry filter not giving Euler angles and making their own. Moral of the story, don't ever use Euler angles :P

Reply:
This was matt trying to compensate pitch/roll in the walk engine